ext/gd/gd.c

```diff
@@ -1562,11 +1562,9 @@ static void _php_image_create_from(INTERNAL_FUNCTION_PARAMETERS, int image_type,
 		pefree(pstr, 1);
 		zend_string_release_ex(buff, 0);
 	}
-	else if (php_stream_can_cast(stream, PHP_STREAM_AS_STDIO)) {
-		/* try and force the stream to be FILE* */
-		if (FAILURE == php_stream_cast(stream, PHP_STREAM_AS_STDIO | PHP_STREAM_CAST_TRY_HARD, (void **) &fp, REPORT_ERRORS)) {
-			goto out_err;
-		}
+	/* try and force the stream to be FILE* */
+	else if (FAILURE == php_stream_cast(stream, PHP_STREAM_AS_STDIO | PHP_STREAM_CAST_TRY_HARD, (void **) &fp, REPORT_ERRORS)) {
+		goto out_err;
```

Comment:
This might cause a change in behavior; if `REPORT_ERRORS` is true, but the casting attempt would fail, we now will raise a warning, while we so far we did not.

Reply:
Yes this is not correct. What I mentioned in that bug - this applies only for cases where cast does not report errors.

Reply:
So this should be probably removed...